dotCMS/src/integration-test/java/com/dotcms/util/FiltersUtil.java

```diff
@@ -85,7 +86,7 @@ public Contentlet createVanityUrl(String title, String site, String uri,
         contentlet.setIndexPolicyDependencies(IndexPolicy.FORCE);
         contentlet.setBoolProperty(Contentlet.IS_TEST_MODE, true);
 
-        return contentlet;
+        return (DefaultVanityUrl) APILocator.getVanityUrlAPI().fromContentlet(contentlet);
```

Comment:
Why changing the return type? You should leave `Contentlet`

Reply:
This is to avoid transforming it when this is called, so we can use the VanityUrl specific methods.